### ISSUE TITLE
Anonymous type to value tuple casting

### ISSUE THREAD

**Issue by mojtabakaviani** [dotnet/roslyn#22889]
With many improvments for value tuple type, now if can casting anonymous type to tuple. a big change in MVC when pass model to view. for example:
return model:
```csharp
var news = from n in db.News 
           select (n.Id, n.Title, CreatedBy = n.User.Name);
return View(news);
```
in view:
```csharp
@model IEnumerable<(int Id, string Title, string CreatedBy)>
```

but before is must solved #12897

**Comment by jcouv** [comment]
I'm sorry, I didn't understand the issue.
Does the code you included in your description work? If not, what error are you getting?
If it's possible, could you describe the problem with pure C# (no MVC)? That would help me understand the problem.
Thanks

**Comment by alrz** [comment]
@mojtabakaviani  I think this is a duplicate of #12897 as you mentioned. The example you gave wouldn't be possible without #12897 (and the corresponding feature in EF (or any other LINQ provider)).

**Comment by mojtabakaviani** [comment]
Yes, anonymous types just access in definition block and can not use in deferent code block. #12897 issue is expression trees not support for tuples, but in want can cast anonymous type to tuples.
```csharp
var news1 = from n in db.News 
           select new { n.Id, n.Title, CreatedBy = n.User.Name };
//casting anonymous to tuple
var news2 = news1 as IEnumerable<(int Id, string Title, string CreatedBy)>;
```
in this simple casting anonymous to tuples and can send to other code block.

**Comment by alrz** [comment]
no type equivalency is defined between a value type and anonymous types. even if we had support for duck typing, that wouldn't work because tuple names are not present in the type itself, (just as metadata).

**Comment by CyrusNajmabadi** [comment]
This is not on roslyn to change.  The language does not consider anonymous types and tuples the same.  If you want that changed, a discussion needs to be opened on dotnet/csharplang.
